source/slang/emit.cpp

```diff
@@ -3959,14 +3959,29 @@ struct EmitVisitor
 
         case kIROp_Select:
             {
-                auto prec = kEOp_Conditional;
-                needClose = maybeEmitParens(outerPrec, prec);
-
-                emitIROperand(ctx, inst->getOperand(0), mode, leftSide(outerPrec, prec));
-                emit(" ? ");
-                emitIROperand(ctx, inst->getOperand(1), mode, prec);
-                emit(" : ");
-                emitIROperand(ctx, inst->getOperand(2), mode, rightSide(prec, outerPrec));
+                if (getTarget(ctx) == CodeGenTarget::GLSL &&
+                    inst->getOperand(0)->getDataType()->op != kIROp_BoolType)
```

Comment:
I almost commented that we should use `mix` in all cases, even for scalars, since `?:` in GLSL is short-circuiting and `kIROp_Select` is not. Then I realized that the logic here in `emit.cpp` will never fold an operation with side effects in as a sub-expression, so that is a non-issue.

That is maybe a subtle point and worth documenting, but I agree with the design choice here to favor `?:` for the scalar case just to leave people with code more like what they wrote.

There is a long-term question for Slang of whether we eventually want to switch to having `?:`, `&&`, and `||` short-circuit and, if so, how to transition to those semantics without too much pain for existing users who might accidentally rely on the fact that they don't short-circuit today.